model/user/user.go

```diff
@@ -208,6 +209,27 @@ func (u *DBUser) IncPatchNumber() (int, error) {
 	return dbUser.PatchNumber, nil
 }
 
+func (u *DBUser) AddFavoritedProject(identifier string) error {
+	if util.StringSliceContains(u.FavoriteProjects, identifier) {
+		return errors.Errorf("cannot add duplicate project '%s'", identifier)
+	}
+	update := bson.M{
+		"$push": bson.M{FavoriteProjectsKey: identifier},
+	}
+	if err := UpdateOne(bson.M{IdKey: u.Id}, update); err != nil {
+		return err
+	}
+
+	before := u.FavoriteProjects
+	u.FavoriteProjects = append(u.FavoriteProjects, identifier)
+
+	err := event.LogUserEvent(u.Id, event.UserEventTypeFavoriteProjectsUpdate, before, u.FavoriteProjects)
+	if err != nil {
+		grip.Alert(errors.Errorf("error logging event for adding '%s' to user favorites", identifier))
```

Comment:
Alert is too high of a level for this since that will send slack messages. I think grip.Error would be the right level